core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushProjectionIntoTableScan.java

```diff
@@ -156,15 +167,35 @@ public Result apply(ProjectNode project, Captures captures, Context context)
             newProjectionAssignments.put(entry.getKey(), replaceExpression(entry.getValue(), nodesToNewPartialProjections));
         });
 
+        Optional<PlanNodeStatsEstimate> newStatistics = tableScan.getStatistics().map(statistics -> {
+            PlanNodeStatsEstimate.Builder builder = PlanNodeStatsEstimate.builder();
+            builder.setOutputRowCount(statistics.getOutputRowCount());
+
+            for (int i = 0; i < connectorPartialProjections.size(); i++) {
+                ConnectorExpression inputConnectorExpression = connectorPartialProjections.get(i);
+                ConnectorExpression resultConnectorExpression = newConnectorPartialProjections.get(i);
+                if (!(resultConnectorExpression instanceof Variable)) {
+                    continue;
+                }
+                String resultVariableName = ((Variable) resultConnectorExpression).getName();
+                Expression inputExpression = ConnectorExpressionTranslator.translate(inputConnectorExpression, inputVariableMappings, new LiteralEncoder(metadata));
+                SymbolStatsEstimate symbolStatistics = scalarStatsCalculator.calculate(inputExpression, statistics, context.getSession(), context.getSymbolAllocator().getTypes());
+                builder.addSymbolStatistics(variableMappings.get(resultVariableName), symbolStatistics);
+            }
+            return builder.build();
+        });
+
         verifyTablePartitioning(context, tableScan, result.get().getHandle());
         return Result.ofPlanNode(
                 new ProjectNode(
                         context.getIdAllocator().getNextId(),
-                        TableScanNode.newInstance(
+                        new TableScanNode(
                                 tableScan.getId(),
                                 result.get().getHandle(),
                                 newScanOutputs,
                                 newScanAssignments,
+                                TupleDomain.all(),
```

Comment:
nit: maybe we can propagate some of the enforced constrains from the original tableScan to the new one...